src/ECharts/Formatter.purs

```diff
@@ -15,10 +16,14 @@ import Control.Monad.Eff
 
 type FormatParams = Json
 
+foreign import data GenericFormatter :: *
+
 data Formatter =
   Template String
   | FormatFunc (forall eff. Array FormatParams -> Eff eff String)
-
+  | StringFormatFunc (String -> String)
+  | NumberFormatFunc (Number -> String)
+  | GenericFormatFunc GenericFormatter
```

Comment:
Actually it is used to incorporate foreign javascript functions, such as:

`exports.toolTipFomatter = function(fn) {
	return	function (params,ticket,callback) {
		var res = params[0].name;
		for (var i = 0, l =  params.length; i < l; i++) {
			res += '<br/>' + params[i].seriesName + ' : ' + fn(params[i].value);		
    	}	
    	return res;
    }
};`

I changed the name a little bit to `ForeignFormatFunc` to make it makes more senses.